RCAIDE/Library/Methods/Energy/Sources/Batteries/Lithium_Ion_NMC/compute_nmc_cell_performance.py

```diff
@@ -98,83 +95,97 @@ def compute_nmc_cell_performance(battery,state,bus,battery_discharge_flag):
-        delta_S               = -496.66*(SOC[t_idx])**6 +  1729.4*(SOC[t_idx])**5 + -2278 *(SOC[t_idx])**4 +  1382.2 *(SOC[t_idx])**3 + \
-                -380.47*(SOC[t_idx])**2 +  46.508*(SOC[t_idx])    + -10.692  
-
-        i_cell                = I_cell[t_idx]/electrode_area # current intensity 
-        q_dot_entropy         = -(T_cell[t_idx])*delta_S*i_cell/(n*F)       
-        q_dot_joule           = (i_cell**2)/sigma                   
-        Q_heat_cell[t_idx]    = (q_dot_joule + q_dot_entropy)*As_cell 
-        Q_heat_pack[t_idx]    = Q_heat_cell[t_idx]*n_total  
-
-        V_ul[t_idx]           = compute_nmc_cell_state(battery_data,SOC[t_idx],T_cell[t_idx],I_cell[t_idx]) 
-
-        V_oc[t_idx]           = V_ul[t_idx] + (I_cell[t_idx] * R_0[t_idx])              
-
-        # Effective Power flowing through battery 
-        P_pack[t_idx]         = P_bat[t_idx]  - np.abs(Q_heat_pack[t_idx]) 
-
-        # store remaining variables 
-        I_pack[t_idx]         = I_bat[t_idx]  
-        V_oc_pack[t_idx]      = V_oc[t_idx]*n_series 
-        V_ul_pack[t_idx]      = V_ul[t_idx]*n_series  
-        T_pack[t_idx]         = T_cell[t_idx] 
-        P_cell[t_idx]         = P_pack[t_idx]/n_total  
-        E_cell[t_idx]         = E_pack[t_idx]/n_total  
-
-        # ---------------------------------------------------------------------------------------------------
-        # Current State 
-        # --------------------------------------------------------------------------------------------------- 
-        if t_idx != state.numerics.number_of_control_points-1:  
-            # Compute cell temperature   
-            HAS_results  = HAS.compute_heat_removed(battery,Q_heat_cell[t_idx],T_cell[t_idx],state,delta_t[t_idx],t_idx) 
-            HEX_results  = HEX.compute_heat_removed(HAS_results,state,delta_t[t_idx],t_idx)
-
-            # Temperature 
-            T_cell[t_idx+1] = HAS_results.current_battery_temperature
-
-            # Compute state of charge and depth of discarge of the battery
-            E_pack[t_idx+1]                          = E_pack[t_idx] -P_pack[t_idx]*delta_t[t_idx] 
-            E_pack[t_idx+1][E_pack[t_idx+1] > E_max] = E_max
-            SOC[t_idx+1]                             = E_pack[t_idx+1]/E_max 
-            SOC[t_idx+1][SOC[t_idx+1]>1]             = 1.
-            SOC[t_idx+1][SOC[t_idx+1]<0]             = 0. 
-            DOD_cell[t_idx+1]                        = 1 - SOC[t_idx+1]  
-
-            # Determine new charge throughput (the amount of charge gone through the battery)
-            Q_cell[t_idx+1]    = Q_cell[t_idx] + I_cell[t_idx]*delta_t[t_idx]/Units.hr  
-
-    return battery    
+    n_total           = battery.pack.electrical_configuration.total
+    
+    # ---------------------------------------------------------------------------------
+    # Examine Thermal Management System
+    # ---------------------------------------------------------------------------------
+    HAS = None  
```

Comment:
See previous comment

Reply:
this for loop helps pass through it even when the coolant_lines is empty.